pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListener.java

```diff
@@ -49,44 +55,55 @@ public class ResourceGroupConfigListener implements Consumer<Notification> {
     private final ResourceGroupResources rgResources;
     private final ResourceGroupNamespaceConfigListener rgNamespaceConfigListener;
 
+    @SneakyThrows
     public ResourceGroupConfigListener(ResourceGroupService rgService, PulsarService pulsarService) {
         this.rgService = rgService;
         this.pulsarService = pulsarService;
         this.rgResources = pulsarService.getPulsarResources().getResourcegroupResources();
-        loadAllResourceGroups();
+        loadAllResourceGroupsAsync()
+                .get(pulsarService.getConfiguration().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS);
```

Comment:
Better not to call with timeout

Reply:
Register the listener when the `loadAllResourceGroups` is complete

Reply:
> Could I use get() or join()?

I think it is worse than `get timeout`

Reply:
Done.